migrations/2_deploy_contracts.ts

```diff
@@ -10,4 +12,11 @@ module.exports = function (deployer) {
   deployer.link(RockPaperScissorsState, RockPaperScissorsGame);
   deployer.link(State, RockPaperScissorsGame);
   deployer.deploy(RockPaperScissorsGame);
+
+  deployer.link(State, Rules);
+  deployer.deploy(Rules);
+  // Linking these without deploying doesn't seem to update the bytecode
+  // so we probably don't have to bother doing it.
+  deployer.link(State, SimpleAdjudicator);
+  deployer.link(Rules, SimpleAdjudicator);
```

Comment:
Linking these beforehand _does_ have an effect in the tests. There the contracts are loaded as:
```js
var RpsStateContract = artifacts.require("./RockPaperScissorsState.sol");
```
It could be that you're not getting the linked version because you're loading the file directly, instead of loading from the truffle "artifact area" - whatever that is.